src/nulog-inference-service/NulogServer.py

```diff
@@ -0,0 +1,88 @@
+# Standard Library
+import logging
+import os
+import time
+from typing import List
+
+# Third Party
+import boto3
+import inference as nuloginf
+from botocore.config import Config
+from NuLogParser import using_GPU
+
+MINIO_ACCESS_KEY = os.environ["MINIO_ACCESS_KEY"]
+MINIO_SECRET_KEY = os.environ["MINIO_SECRET_KEY"]
+
+
+class NulogServer:
+    def __init__(self, method: str = "predict"):
+        self.method = method
+        self.to_reload_model = False
+        self.download_from_minio()
+        self.load()
+
+    def download_from_minio(
+        self,
+        decoded_payload={
+            "bucket": "nulog-models",
+            "bucket_files": {
+                "model_file": "nulog_model_latest.pt",
+                "vocab_file": "vocab.txt",
+            },
+        },
+    ):
+
+        endpoint_url = "http://minio.default.svc.cluster.local:9000"
+        minio_client = boto3.resource(
+            "s3",
+            endpoint_url=endpoint_url,
+            aws_access_key_id=MINIO_ACCESS_KEY,
+            aws_secret_access_key=MINIO_SECRET_KEY,
+            config=Config(signature_version="s3v4"),
+        )
+        if not os.path.exists("output/"):
+            os.makedirs("output")
+
+        bucket_name = decoded_payload["bucket"]
+        bucket_files = decoded_payload["bucket_files"]
+        for k in bucket_files:
+            try:
+                minio_client.meta.client.download_file(
+                    bucket_name, bucket_files[k], "output/{}".format(bucket_files[k])
+                )
+            except Exception as e:
+                logging.error(
+                    "Cannot currently obtain necessary model files. Exiting function"
+                )
+                return
+
+    def load(self):
+
+        if using_GPU:
+            logging.info("inferencing with GPU.")
+        else:
+            logging.info("inferencing without GPU.")
+        try:
+            self.parser = nuloginf.init_model()
+            self.nuloginf = nuloginf
+            logging.info("Nulog model gets loaded.")
+            self.to_reload_model = False
+        except Exception as e:
+            logging.error("No Nulog model currently {}".format(e))
+        # self.predict(test_texts)
+
+    def predict(self, logs: List[str], feature_names=None):
+        if self.to_reload_model:
+            self.load()
+        start_time = time.time()
+        output = self.nuloginf.predict(self.parser, logs)
+        logging.info(
+            (
+                "--- predict %s logs in %s seconds ---"
+                % (len(logs), time.time() - start_time)
+            )
+        )
+        return output
```

Comment:
Pls add a check to see if model exists
`self.nuloginf` is not guaranteed to exist if model is not in Minio - which is a use case for any fresh aiops cluster installation
These are logs that I got in this scenario:
```
2021-04-06 02:04:21,942 - ERROR - Cannot currently obtain necessary model files. Exiting function
2021-04-06 02:04:21,942 - INFO - inferencing with GPU.
2021-04-06 02:04:21,942 - INFO - initializing...
2021-04-06 02:04:21,947 - ERROR - No Nulog model currently [Errno 2] No such file or directory: 'output/vocab.txt'
2021-04-06 02:04:21,953 - INFO - Connected to NATS at nats_client:VfU6TcAl9x@nats-client.default.svc:4222...
Traceback (most recent call last):
  File "start-nulog-inference.py", line 135, in <module>
    start_inference_controller()
  File "start-nulog-inference.py", line 127, in start_inference_controller
    loop.run_until_complete(asyncio.gather(inference_coroutine, consumer_coroutine))
  File "/usr/local/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
    return future.result()
  File "start-nulog-inference.py", line 69, in infer_logs
    predictions = nulog_predictor.predict(masked_log)
  File "/app/NulogServer.py", line 78, in predict
    output = self.nuloginf.predict(self.parser, logs)
AttributeError: 'NulogServer' object has no attribute 'nuloginf'
```

Reply:
good catch. issue addressed.